sql2.txt

```diff
@@ -17,29 +31,54 @@ may have ordered the same product in several different orders.  So the total qua
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
 
-
+SELECT c.CustomerName, p.ProductName, SUM(od.Quantity) AS TotalQuantity
+FROM Customers c
+JOIN Orders o
+ON c.CustomerID = o.CustomerID
+JOIN OrderDetails od
+ON o.OrderID = od.OrderID
+JOIN Products p
+ON od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID;
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
 
+SELECT c.CustomerName, p.ProductName, SUM(od.Quantity) AS TotalQuantity, (SUM(od.Quantity) * p.Price) AS TotalPrice
+FROM Customers c
+JOIN Orders o
+ON c.CustomerID = o.CustomerID
+JOIN OrderDetails od
+ON o.OrderID = od.OrderID
+JOIN Products p
+ON od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
 
+INSERT INTO Orders ("CustomerID", "EmployeeID", "OrderDate", "ShipperID")
+VALUES (34, 3, "7/4/1996 ", 3);
 
 6. Add three OrderDetails.  These should specify the ID of the new Order you just created.  For each,
 you will have to specify the ID of an existing product.  Paste your SQL statement below.
 
+INSERT INTO OrderDetails ("OrderID", "ProductID", "Quantity")
+VALUES (10444, 5, 3), (10444, 4, 54), (10444,  6, 20);
 
 7.  Delete the third of the OrderDetails you just added.  Paste your SQL statement below.
 
-
+DELETE FROM OrderDetails 
+WHERE OrderDetailsID == 521;
```

Comment:
Couple things: The column name doesn't have an `s`, as executing the statement results in a "no such column: OrderDetailsID" error. Secondly, while the `==` operator still works in SQLite, you want to be careful with it because other flavors of SQL don't support it, such as Postgres, Oracle, and SQL Server. In my experience with SQL, the convention is simply to use `=`.

```suggestion
WHERE OrderDetailID = 521;
```